### ISSUE TITLE
Why doesn't my model converge？

### ISSUE THREAD

**Issue by michelleqyhqyh** [facebookresearch/fairseq#4280]
### Before asking:
1. search the issues.
2. search the docs.

How to make my model converge?

#### What is your question?
I used this code to train my model:
CUDA_VISIBLE_DEVICES=6 fairseq-train data/top12 --save-dir ./train/top12 --tensorboard-logdir ./train/top12/tensorboard-log --log-format simple --log-interval 100  --save-interval-updates 5000 --arch transformer --optimizer adam --criterion label_smoothed_cross_entropy --seed 3435 --fixed-validation-seed 1 --lr 0.002 --min-lr 1e-09 --lr-scheduler reduce_lr_on_plateau --max-epoch 25 --batch-size 4096 --clip-norm 0.1 --dropout 0.1 --max-tokens 4096 2>&1 | tee ./train/top12/top12-screen.log

But ,it doesn't coverge. This is the output:
![image](https://user-images.githubusercontent.com/45490378/158717663-282ab507-f49f-4350-bcbd-643eb96974fe.png)

By the way, I used sentencepiece to do BPE before I did fairseq-preprocess. The fairseq-preprocess code is :
fairseq-preprocess --source-lang zh --target-lang en --trainpref ./data/top12/original/train --validpref ./data/top12/original/valid --testpref ./data/top12/original/test  --destdir ./data/top12

This is my train file which is processed by tokenize and BPE(sentencepiece) before I did fairseq-preprocess:
![image](https://user-images.githubusercontent.com/45490378/158718525-12e57855-3540-443c-b30d-1a7401185ef4.png)
![image](https://user-images.githubusercontent.com/45490378/158718547-ac18d82f-6edf-41c5-be5e-6f549c14366e.png)

#### What's your environment?

 - fairseq Version (e.g., 1.0 or main): 0.10.2
 - PyTorch Version (e.g., 1.0): 1.8.1
 - OS (e.g., Linux): Linux
 - How you installed fairseq (`pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.7
 - CUDA/cuDNN version: 11
 - GPU models and configuration:
 - Any other relevant information:

**Comment by kkeleve** [comment]
maybe you can try to reduce lr

**Comment by gmryu** [comment]
Can you also share a few about the preprocess log, do you see a lot of `<unk>`?
You may also check how fairseq-preprocess work by editing `fairseq/fairseq/data/dictionary.py `. 
Find the `def encode_line` , which deals with all of those tokens and decide what tensors are written in the .bin.

Then you may just want to change learning rate and scheduler.
(By the way, what is your initial loss? )
(Also, p.s. you do not need \<bos\> and \<eos\>, fairseq has its own bos:\<s\> and eos:\</s\> and automatically append them in fairseq-train.
